silhouette/Graphtec.py

```diff
@@ -686,21 +684,25 @@ def get_tool_setup(self):
       pass
     return 'none'
 
-  def wait_for_ready(self, timeout=30, verbose=True):
+  def wait_for_ready(self, timeout=30, poll_interval=2.0, verbose=False):
     # get_version() is likely to timeout here...
     # if verbose: print("device version: '%s'" % s.get_version(), file=sys.stderr)
     state = self.status()
-    for i in range(1, int(timeout*.5)):
+    if self.dry_run:
+      # not actually sending commands, so don't really care about being ready
+      return state
+    npolls = int(timeout/poll_interval)
+    for i in range(1, npolls):
       if (state == 'ready'): break
       if (state == 'None'):
         raise NotImplementedError("Waiting for ready but no device exists.")
-      if verbose: print(" %d/%d: status=%s\r" % (i, int(timeout*.5), state), end='', file=sys.stderr)
+      if verbose: print(" %d/%d: status=%s\r" % (i, npolls, state), end='', file=sys.stderr)
       if verbose == False:
         if state == 'unloaded':
-          print(" %d/%d: please load media ...\r" % (i, int(timeout*.5)), end='', file=sys.stderr)
-        elif i > 5:
+          print(" %d/%d: please load media ...\r" % (i, npolls, state), end='', file=sys.stderr)
+        elif i > npolls/3:
           print(" %d/%d: status=%s\r" % (i, int(timeout*.5), state), end='', file=sys.stderr)
```

Comment:
did you intentionally leave this untouched or would you want to change?
`- int(timeout*.5)`
`+ npolls`

Reply:
Great catch, thanks, dunno how I missed that one, sorry. Should be fixed now.